### PR TITLE
Added: Longest Common Prefix Program

### LINKED ISSUE #1804: [NEW ALGORITHM] Longest Common Prefix Under Strings
## Issue will be closed if:

> 1) You mention more than one algorithm. You can create a separate issue for each algorithm once the current one is completed.  
> 2) You propose an algorithm that is already present or has been mentioned in a previous issue.  
> 3) You create a new issue without completing your previous issue.

**Note:** These actions will be taken seriously. Failure to follow the guidelines may result in the immediate closure of your issue.

---

### Name:  
```[Longest Common Prefix]```  

### About:  
This program finds the Longest Common Prefix (LCP) among an array of strings. It starts by assuming the first string as the prefix and iteratively compares it with each subsequent string, updating the prefix to the common portion. If no common prefix exists, it stops early. The program is useful for applications like string analysis, autocomplete systems, and pattern matching.

**Input**:
```
Enter the number of strings: 3
Enter the strings:
String 1: flower
String 2: flow
String 3: flight
```

**Output**:
```
The longest common prefix is: fl
```

---

### Labels:  
```new algorithm, gssoc-ext, hacktoberfest, level1```

---

### Assignees:  
- [X] Contributor in GSSoC-ext
- [X] Want to work on it

### PR DESCRIPTION
**Pull request by rajdeepchakraborty-rc**
# Issue Resolved

- Closes #1804 

# Algorithm Added
**Longest Common Prefix**

# Description
This program finds the Longest Common Prefix (LCP) among an array of strings. The LCP is the longest sequence of characters shared at the beginning of all strings in the array. The program employs an efficient algorithm to identify the common prefix by comparing the strings character by character.

```
Example:
Enter the number of strings: 5
Enter the strings:
String 1: prawn
String 2: prajakta
String 3: prince
String 4: probably
String 5: project

Output:
The longest common prefix is: pr
```

### CODE REVIEW

**Comment by rajdeepchakraborty-rc** [comment]
@pankaj-bind Please review the PR at your convinience

**Comment by rajdeepchakraborty-rc** [comment]
Also, after the recent PR, you merged: 
- PR #1821

The Contributor has again broken the codebase cloning due to improper directory path naming:
image

The space in the Folder name is failing the cloning process:

Screenshot 2024-11-10 021215

**Review by pankaj-bind — APPROVED** [review]

**Comment by rajdeepchakraborty-rc** [comment]
@pankaj-bind please do assign the PR to me. I had opened an unmerged PR request as instructed by the Core team.

**Pull request merged**
